src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -907,6 +907,76 @@ public override IOperation VisitWhileLoop(IWhileLoopOperation operation, int? ca
+        {
+            foreach (IVariableDeclaratorOperation declarator in operation.Declarators)
+            {
+                ILocalSymbol localSymbol = declarator.Symbol;
+
+                IOperation initializer = null;
+                SyntaxNode assignmentSyntax = null;
+                if (declarator.Initializer != null)
+                {
+                    initializer = Visit(declarator.Initializer.Value, captureIdForResult);
+                    assignmentSyntax = declarator.Syntax;
+                }
+
+                if (operation.Initializer != null)
+                {
+                    IOperation operationInitializer = Visit(operation.Initializer.Value, captureIdForResult);
+                    assignmentSyntax = operation.Syntax;
+                    if (initializer != null)
+                    {
+                        initializer = new InvalidOperation(ImmutableArray.Create(initializer, operationInitializer), semanticModel: null, operation.Syntax, type: localSymbol.Type, constantValue: default, isImplicit: true);
+                    }
+                    else
+                    {
+                        initializer = operationInitializer;
+                    }
+                }
+
+                if (initializer != null)
+                {
+                    // We can't use the IdentifierToken as the syntax for the local reference, so we use the entire declarator as the node
+                    var localRef = new LocalReferenceExpression(localSymbol, isDeclaration: true, semanticModel: null, declarator.Syntax, localSymbol.Type, constantValue: default, isImplicit: true);
+                    // prototype(dataflow): We'd like to remove ExpressionStatements from the CFG altogether, as they're useless when all you need to do is look to see if the parent is null
+                    var statement = new ExpressionStatement(new SimpleAssignmentExpression(localRef, isRef: localSymbol.IsRef, initializer, semanticModel: null, assignmentSyntax, localRef.Type, constantValue: default, isImplicit: true),
+                                                            semanticModel: null, assignmentSyntax, type: null, constantValue: default, isImplicit: true);
+                    AddStatement(statement);
+                }
+            }
+        }
+
+        public override IOperation VisitVariableDeclaration(IVariableDeclarationOperation operation, int? captureIdForResult)
+        {
+            // All variable declarators should be handled by VisitVariableDeclarationGroup.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        public override IOperation VisitVariableDeclarator(IVariableDeclaratorOperation operation, int? captureIdForResult)
+        {
+            // All variable declarators should be handled by VisitVariableDeclaration.
+            // prototype(dataflow): try/catch can get directly here, as can foreach control variables
```

Comment:
>// prototype(dataflow): try/catch can get directly here, as can foreach control variables [](start = 12, length = 89)

Please remove this comment. try/catch won't get here once we handle it properly. No adjustments will  be necessary here.

Reply:
Ok, removed.

---
In reply to: [167021700](https://github.com/dotnet/roslyn/pull/24692#discussion_r167021700) [](ancestors = 167021700)